client/src/cmdhfseos.c

```diff
@@ -526,7 +526,7 @@ static void create_mutual_auth_key(uint8_t *KEYIFD, uint8_t *KEYICC, uint8_t *RN
         counter++;
         set_counter_big_endian(hash_in, counter);
         sha1hash(hash_in, sizeof(hash_in), hashedOutput);
-        memcpy(output + 20, hashedOutput, 20);
+        memcpy(output + 20, hashedOutput, 12);
```

Comment:
12 ?    you sure you didn't mean 16?

Reply:
Yes, only 32 bytes are used total. SHA1 outputs 20 bytes each time, 40 total, but it just takes the first 32 and ignores the rest.

Reply:
good, 
so `hashedOutput[32]`  could be [20]  ?